tools/memap.py

```diff
@@ -396,38 +397,48 @@ def generate_output(self, export_format, file_output=None):
         """
 
         try:
-            if file_output:
-                file_desc = open(file_output, 'wb')
+            if silent:
+                file_desc = None
             else:
-                file_desc = sys.stdout
+                if file_output:
+                    file_desc = open(file_output, 'wb')
+                else:
+                    file_desc = sys.stdout
```

Comment:
A bit too much much if->if->then->else. I understand that the new code requires that memap generates string output, not write to files, therefore I'd expect that the memap is split into:
* methods that generate string output
* methods that actually write or print output

Not spaghetti mix of these